src/pyskindose/format_export_data.py

```diff
@@ -0,0 +1,475 @@
+
+    def _extract_position_list(self, phantom: Phantom, data_norm: pd.DataFrame) -> list[Position]:
+        return [
+            self._get_position_dict(phantom=phantom, data_norm=data_norm, event=ind) for ind in range(len(data_norm))
+        ]
+
+    @staticmethod
+    def _get_position_dict(phantom: Phantom, data_norm: pd.DataFrame, event: int) -> Position:
+        phantom.position(data_norm=data_norm, event=event)
+        return Position(
+            x=phantom.r[:, 0].tolist(),
+            y=phantom.r[:, 1].tolist(),
+            z=phantom.r[:, 2].tolist(),
+        )
+
+    @staticmethod
+    def _extract_beam_data_list(
+        data_norm: pd.DataFrame, event: int
+    ) -> tuple[Position, VertexIndices, Position, VertexIndices]:
+        beam = Beam(data_norm, event=event, plot_setup=False)
+        beam_position = Position(x=beam.r[:, 0].tolist(), y=beam.r[:, 1].tolist(), z=beam.r[:, 2].tolist())
+        beam_vertex_indices = VertexIndices(
+            i=beam.ijk[:, 0].tolist(), j=beam.ijk[:, 1].tolist(), k=beam.ijk[:, 2].tolist()
+        )
+        detector_position = Position(
+            x=beam.det_r[:, 0].tolist(), y=beam.det_r[:, 1].tolist(), z=beam.det_r[:, 2].tolist()
+        )
+        detector_vertex_indices = VertexIndices(
+            i=beam.det_ijk[:, 0].tolist(), j=beam.det_ijk[:, 1].tolist(), k=beam.det_ijk[:, 2].tolist()
+        )
+
+        return beam_position, beam_vertex_indices, detector_position, detector_vertex_indices
+
+    def to_dict(self):
+        return {
+            "number_of_events": self.events,
+            "rotation": self.rotation,
+            "translation": self.translation,
+            "phantom_object_trace_order": self.phantom_object_trace_order,
+            "beam": {
+                "positions": [pos.to_dict() for pos in self.beam_positions],
+                "vertex_indices": [pos.to_dict() for pos in self.beam_vertex_indices],
+                "trace_order": self.beam_wireframe_trace_order,
+            },
+            "detector": {
+                "positions": [pos.to_dict() for pos in self.beam_positions],
+                "vertex_indices": [pos.to_dict() for pos in self.beam_vertex_indices],
+                "trace_order": self.detector_wireframe_trace_order,
+            },
+        }
+
+
+class PySkinDoseOutput:
+    """A collection of the information resulting from the PySkinDose analysis
+
+    Attributes
+    __________
+
+    PSD : float
+        The peak skin dose found in the dose map
+    AirKerma : float
+        The total air KERMA of the
+    Events : EventOutput
+        The event data for the examination
+    Patient : dict[str, str | HumanPhantomOutput | NonHumanPhantomOutput]
+        Information on the patient used in the calculations
+    PatientOffsets : dict[str, float]
+        The base offsets in the long, vert, and lat direction
+    Table : Phantom
+        The treatment table as an instance of the Phantom class
```

Comment:
we are using "patient support table" on other places, but i like treatment table more... we need to to define a variable/parameter convention.